wrangler-core/src/main/java/co/cask/directives/date/FormatDate.java

```diff
@@ -76,21 +78,32 @@ public void destroy() {
     for (Row row : rows) {
       Row dt = new Row(row);
       int idx = dt.find(column);
-      if (idx != -1) {
-        Object object = row.getValue(idx);
-        if (object != null && object instanceof Date) {
-          dt.setValue(idx, destinationFmt.format((Date) object));
-        } else {
+
+      if (idx == -1) {
+        throw new DirectiveExecutionException(toString() + " : '" + column + "' column is not defined in the row. " +
+                                                "Please check the wrangling step."
+        );
+      }
+
+      Object object = row.getValue(idx);
+
+      if (object != null) {
+        try {
+          if (object instanceof LocalDate) {
+            object = ((LocalDate) object).atStartOfDay(ZoneId.systemDefault());
```

Comment:
instead of writing over the value of object, it's cleaner and more readable to have a separate ZonedDateTime variable.

Reply:
should always use UTC. From the avro spec, a date is the number of days since the unix epoch.

Reply:
seems like this should be UTC, not the system default. In general, it makes things more predictable if the behavior doesn't change depending on the environment it runs on.